src/lib/fetch.ts

```diff
@@ -4,20 +4,30 @@ import { URL } from 'url';
 import * as fs from 'fs';
 import { readFile } from 'fs/promises';
 
-export async function azleFetch(input: any, init?: any): Promise<any> {
+export async function azleFetch(
+    input: RequestInfo | URL,
+    init?: RequestInit | undefined
+): Promise<Response> {
     if (process.env.AZLE_TEST_FETCH === 'true') {
         console.log('azleFetch has been called');
     }
 
+    if (typeof input !== 'string') {
+        throw new Error(`azleFetch input can only be of type string`);
+    }
+
     const url = new URL(input);
 
     if (url.protocol === 'icp:') {
         const canisterId = url.hostname;
         const canisterMethod = url.pathname.replace('/', '');
 
         const { method, headers, body } = init ?? {};
```

Comment:
It has come to my attention that headers is not used after this point. That seems like a problem to me

Reply:
Can you underscore these to signify that they are unused on purpose

Reply:
I just removed them